gotestcov

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -eu
+
+# find_go_mod walks up the directory tree looking for the go.mod file.
+# If it doesn't find it, the script will be aborted.
+find_go_mod() {
+    cwd="$(pwd)"
+
+    while [ "$cwd" != "/" ]; do
+        if [ -f "$cwd/go.mod" ]; then
+            echo "$cwd"
+            return
+        fi
+        cwd=$(dirname "$cwd")
+    done
+    echo "Error: go.mod not found in parent path. Aborting!"
+    exit 1
+}
+
+projectroot="$(find_go_mod)"
+cov_dir="${projectroot}/coverage"
+mkdir -p "${cov_dir}"
+
+# start http server on 6061 if none
+if ! $(nc -z localhost 6061); then
+    nohup python3 -m http.server --directory "${cov_dir}" 6061 1>/dev/null 2>&1 &
+fi
+
+raw_cov_dir="${cov_dir}/raw_files"
+
+mkdir -p "${raw_cov_dir}"
+
+# Clean the previous coverage data generated
+if [ -e "${raw_cov_dir}/" ]; then
+    rm -vfr "${raw_cov_dir}/*"
```

Comment:
This test will always pass since the directory has been created on previous line.